cuBERT_topic_modelling/cuBERTopic.py

```diff
@@ -0,0 +1,563 @@
+
+            # Calculate new topic words
+            tf_idf, count = self.new_c_tf_idf(docs_df, len(docs_df))
+            top_n_words, name_repr = self.extract_top_n_words_per_topic(
+                tf_idf, count, docs_per_topics_topics, n=30
+            )
+
+            return docs_df, top_n_words
+
+    def fit_transform(self, data):
+        """Fit the models on a collection of documents, generate topics, and return
+        the docs with topics
+        Arguments:
+            documents: A list of documents to fit on
+
+        Returns:
+            predictions: Topic predictions for each documents
+            probabilities: The probability of the assigned topic per document.
+        """
+
+        documents = cudf.DataFrame(
+            {"Document": data, "ID": range(len(data)), "Topic": None}
+        )
+
+        # Extract embeddings
+        embeddings = self.create_embeddings(
+            documents.Document
+        )
+
+        # Reduce dimensionality with UMAP
+        umap_embeddings = self.reduce_dimensionality(embeddings)
+        del embeddings
+
+        # Cluster UMAP embeddings with HDBSCAN
+        documents, probabilities = self.clustering_hdbscan(
+            umap_embeddings,
+            documents
+        )
+
+        del umap_embeddings
+
+        documents = self.sort_mappings_by_frequency(documents)
+
+        tf_idf, count, docs_per_topics_topics, docs_df = self.create_topics(
+            documents
+        )
+        top_n_words, name_repr = self.extract_top_n_words_per_topic(
+            tf_idf, count, docs_per_topics_topics, n=30
+        )
+
+        self.topic_sizes_df["Name"] = self.topic_sizes_df["Topic"].map(
+            name_repr
+        )
+        self.top_n_words = top_n_words
+        predictions = documents.Topic.to_arrow().to_pylist()
+
+        return (predictions, probabilities)
+
+    def get_topic(self, topic, num_words=10):
+        """Return top n words for a specific topic and their c-TF-IDF scores
+        Arguments:
+            topic: A specific topic for which you want its representation
+            num_words: Number of words we want in the representation
+        Returns:
+            The top n words for a specific word and its respective
+            c-TF-IDF scores
+        """
+        return self.top_n_words[int(self.final_topic_mapping[topic])][:num_words]
+
+    def get_topic_info(self):
+        """Get information about each topic including its id, frequency, and name
+
+        Returns:
+            info: The information relating to all topics
+        """
+
+        # Note: getting topics in sorted order without using
+        # TopicMapper, as in BERTopic
+        topic_sizes_df_columns = self.topic_sizes_df.Name.str.split(
+            "_", expand=True
+        )[[0, 1, 2, 3, 4]]
+        self.original_topic_mapping = topic_sizes_df_columns[0]
+        self.new_topic_mapping = sorted(
+            self.topic_sizes_df["Topic"].to_pandas()
+        )
+
```

Comment:
We dont need to convert to_pandas here. Come up with a solution that works without `to_arrow ` call please.